### ISSUE TITLE
Add resource `okta_branding` for brand customization deployment

### ISSUE THREAD

**Issue by frousselet** [okta/terraform-provider-okta#988]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Hello,

We would like to be able to customize Okta via Terraform.
You will find bellow a potential Terraform configuration based on the API : https://developer.okta.com/docs/reference/api/brands/

### New or Affected Resource(s)

- __New resource:__ `okta_branding`
- __Affected resource:__ `okta_org_configuration` (flag `logo` as `OBSOLETE`)

### Potential Terraform Configuration

```hcl
resource "okta_branding" "global" {
  primary_color_hex            = "#1662dd"
  primary_color_contrast_hex   = "#ffffff" # Or #000000
  secondary_color_hex          = "#ebebed"
  secondary_color_contrast_hex = "#000000" # Or #ffffff

  signin_page_touch_point_variant        = "OKTA_DEFAULT" # Or BACKGROUND_SECONDARY_COLOR / BACKGROUND_IMAGE
  end_user_dashboard_touch_point_variant = "OKTA_DEFAULT" # Or WHITE_LOGO_BACKGROUND / FULL_THEME / LOGO_ON_FULL_WHITE_BACKGROUND
  error_page_touch_point_variant         = "OKTA_DEFAULT" # Or BACKGROUND_SECONDARY_COLOR / BACKGROUND_IMAGE
  email_template_touch_point_variant     = "OKTA_DEFAULT" # Or FULL_THEME

  logo             = file("assets/logo.png") # Replace okta_org_configuration / logo
  favicon          = file("assets/favicon.ico")
  background_image = file("assets/background.png")

  remove_powered_by_okta = true # Default is false

  agree_to_custom_privacy_policy = true # Default is false, fails if custom_privacy_policy_url is absent
  custom_privacy_policy_url      = "https://example.com/privacy"
}
```

### References

- #895

**Comment by monde** [comment]
@frousselet we are planning to update the provider with all the management API features that have been exposed in okta/okta-sdk-golang . See #979. That work will cover this.

**Comment by monde** [comment]
@frousselet the latest release of the provider includes these data sources and resources for brands and branded email customization. I'm working on the theme operations of the brands API this sprint.

  * Data Sources
    * `okta_brands`
    * `okta_brand`
    * `okta_email_customizations`
    * `okta_email_customization`
    * `okta_email_templates`
    * `okta_email_template`
  * Resources
    * `okta_brand`
    * `okta_email_customization`

**Comment by frousselet** [comment]
Hi @monde,

Thanks for the update!

What are the differences between data sources with an `s` and those without?

**Comment by monde** [comment]
Released v3.27.0, it includes:
  * Data Sources
    * `okta_themes`
    * `okta_theme`
  * Resources
    * `okta_theme`

Please check the documentation. All of the data sources and resources now exists to produce this behavior, but in a more flexible manner. @frousselet the example resource in this initial request `resource "okta_branding"` is describing what is available in the [Themes API](https://developer.okta.com/docs/reference/api/brands/#themes-operations) and are expressed in our theme(s) data sources and theme resource. The greater [Brands API](https://developer.okta.com/docs/reference/api/brands/) includes the brand objects, email templates/customization, as well themes.

- https://registry.terraform.io/providers/okta/okta/latest/docs
- https://registry.terraform.io/providers/okta/okta/latest/docs/data-sources/themes
- https://registry.terraform.io/providers/okta/okta/latest/docs/data-sources/theme
- https://registry.terraform.io/providers/okta/okta/latest/docs/resources/theme
